src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/CmsSignature.cs

```diff
@@ -142,6 +143,22 @@ internal bool Sign(
             return signed;
         }
 
+        private static IDisposable? GetSigningKey<T>(object? privateKey, X509Certificate2 certificate, bool silent, Func<T?> getCertPublicKey, out T? signingKey)
```

Comment:
Looks like callers of this are now going to be allocating a delegate to pass in as getCertPublicKey, e.g.
```C#
using (GetSigningKey(key, certificate, silent, certificate.GetDSAPublicKey, out DSA? dsa))
```
Should this use `Func<TState, T?>` and accept a `TState` argument to thread through in order to avoid that allocation?